src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageDependencies.cs

```diff
@@ -119,6 +119,8 @@ public string ProjectLanguage
         /// </summary>
         public bool EmitLegacyAssetsFileItems { get; set; } = false;
 
+        public string TargetFrameworkMoniker { get; set; }
```

Comment:
only Diagnostic is relevant to TargetFrameworkMoniker? Should "PackageDependencies" be different when TargetFrameworkMoniker is different? (Although "Diagnostic" is the only thing TargetFrameworkMoniker applied to.)

Reply:
If so, maybe we should call it something like `DiagnosticTargetFrameworkMoniker`

Reply:
The input items have already filtered by `TargetFrameworkMoniker` in `ResolvePackageDependencies`. I like keeping this property name the same as the MSBuild property that it comes from. The fact that it's currently only used for diagnostics is kind of an implementation detail.